metpy/plots/skewt.py

```diff
@@ -416,6 +416,30 @@ def plot_barbs(self, p, u, v, c=None, xloc=1.0, x_clip_radius=0.1,
         b.set_clip_box(transforms.TransformedBbox(ax_bbox, self.ax.transAxes))
         return b
 
+    def get_t0(self, spacing=10):
+        r"""
+        
+        Return range of temperatures based on xaxis
+        Parameters
+        ----------
+        spacing : int or float, optional
+            Space between returned temperature values
+        
+        Returns
+        -------
+        units.Quantity 
+        """
+        # minimum and maximum temperature
+        xmin, xmax = self.ax.get_xlim()
+        
+        # units if they were supplied are stored in the xaxis
+        tunits = self.ax.xaxis.units
+        
+        # if temperature has no units, guess from the lower bound
+        if tunits is None:
```

Comment:
Nice way to handle this. Curious though if we just require units on pressure and temperature, we can avoid extra conditionals like this?

Reply:
I am of the opinion that requiring units is better than having unit checks sprinkled throughout the code

Reply:
Thoughts @dopplershift? Don't want to make @jibbals go about adding the decorators before coming to a decision on this overarching question first.

Reply:
There's a `set_units` call for the x and y axis. Is there any reason we shouldn't just set those when we create the SkewT? That feels like the unit-correct thing to do, alongside setting some sane default limits. I have a PR going in that can set the units if we think that's a good way to proceed.